src/App.js

```diff
@@ -2,12 +2,64 @@ import React from 'react';
 
 import './App.scss';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { GoodsList } from './components/GoodsList';
 
-const App = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+class App extends React.Component {
+  state = {
+    goods: [],
+  };
+
+  showAll = () => {
+    getAll()
+      .then((goods) => {
+        this.setState({ goods });
+      });
+  }
+
+  showFiveFirst = () => {
+    get5First()
+      .then((goods) => {
+        this.setState({ goods });
+      });
+  }
+
+  showRedGoods = () => {
+    getRedGoods()
+      .then((goods) => {
+        this.setState({ goods });
+      });
+  }
```

Comment:
Your functions do the same.
I think you should create only one function and accept different callback for buttons.

Reply:
agree, but now my code looks much better and clear